### PR TITLE
Remove invalid promise catches.

### LINKED ISSUE #906: Remove invaild catches from replayer
[These catches](https://github.com/ckeditor/ckeditor5-engine/pull/902/files/dd3015b82786b648e0c77f949c79d8a0cace4ca3#diff-891fb910d1bb2fa1a25f134274923658R103) should be changed in order to provide some correct errors.

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Removed invalid promise catches from replayer tool. Closes #906.

### CODE REVIEW

File: src/dev-utils/deltareplayer.js [hunk 1]
```diff
@@ -99,24 +104,28 @@ export default class DeltaReplayer {
...
+				}
+			} );
 	}
 
 	/**
 	 * Applies the next delta to replay.
+	 * Returns promise with a `isFinished` parameter.
 	 *
-	 * @returns {Promise}
+	 * @returns {Promise.<Boolean>}
```
Comment (scofalik) [inline src/dev-utils/deltareplayer.js]: I know that you were doing documentation refactoring so you should know better but I have to ask - do we use such notation somewhere and does it/will it correctly show in built docs? I see it first time :P

File: src/dev-utils/deltareplayer.js [hunk 2]
```diff
@@ -99,24 +104,28 @@ export default class DeltaReplayer {
...
+			.then( ( isFinished ) => {
+				if ( !isFinished ) {
+					return this.applyAllDeltas();
+				}
+			} );
 	}
 
 	/**
 	 * Applies the next delta to replay.
+	 * Returns promise with a `isFinished` parameter.
```
Comment (scofalik) [inline src/dev-utils/deltareplayer.js]: an `isFinished`

File: src/dev-utils/deltareplayer.js [hunk 3]
```diff
@@ -99,24 +104,28 @@ export default class DeltaReplayer {
...
 	 *
-	 * @returns {Promise}
+	 * @returns {Promise.<Boolean>}
 	 */
 	applyNextDelta() {
 		const document = this._document;
```
Comment (scofalik) [inline src/dev-utils/deltareplayer.js]: I know that we've been discussing moving here this check:
```javascript
if ( !jsonDelta ) {
return Promise.resolve();
}
```

Was there a reason to abandon this idea?

Comment (ma2ciek) [inline src/dev-utils/deltareplayer.js]: I think it's a safer way to do it. If we would want to change this class in the future and add some other `enqueuChanges()` modifying `this._deltasToReplay` then some strange errors could occur. (As far as I understand the idea behind `enqueueChanges` ;)

File: src/dev-utils/deltareplayer.js (hunk 1 above)
Comment (ma2ciek) [inline src/dev-utils/deltareplayer.js]: Yes, we do, e.g. here: https://github.com/ckeditor/ckeditor5-image/blob/0e8c6cab3759f5616f6d02a9ffc31aa099491adc/src/imagetextalternative.js#L108. The generic types are documented here: http://usejsdoc.org/tags-type.html, I didn't create the separate doc about them (maybe I should (?)). They're building correctly. And here is a list of currently supported generic types: https://github.com/ckeditor/ckeditor5-dev/blob/0aa523b441a052a441005041af9df4b7bd493c0e/packages/jsdoc-plugins/lib/validator/types.js

File: tests/dev-utils/deltareplayer.js [hunk 4]
```diff
@@ -149,6 +131,26 @@ describe( 'DeltaReplayer', () => {
...
+				.map( d => JSON.stringify( d ) )
+				.join( '---' );
+
+			const deltaReplayer = new DeltaReplayer( doc, '---', stringifiedDeltas );
+
+			return deltaReplayer.play( 1 )
+				.then( () => {
+					throw new Error( 'It should throw an error' );
+				}, ( err ) => {
+					expect( err.name ).to.equal( 'CKEditorError' );
```
Comment (Reinmar) [inline tests/dev-utils/deltareplayer.js]: It's worth checking the message. We usually check it with `expect( err.message ).to.match( /^the-error-identifier:/ )` – this is precise enough, but you're also sure that you caught the right error. Not any CKEditorError.

**Pull request merged**